packages/lib/services/plugins/RepositoryApi.ts

```diff
@@ -165,24 +191,62 @@ export default class RepositoryApi {
 		}
 	}
 
-	public async search(query: string): Promise<PluginManifest[]> {
-		query = query.toLowerCase().trim();
+	public async searchAndFilter(category?: string, searchQuery?: string): Promise<PluginManifest[]> {
+		const manifests: PluginManifest[] = await this.manifests();
+		let output: PluginManifest[] = [];
+		const pluginStates = Setting.value('plugins.states');
+		category = category.toLowerCase();
 
-		const manifests = await this.manifests();
-		const output: PluginManifest[] = [];
-
-		for (const manifest of manifests) {
-			for (const field of ['name', 'description']) {
-				const v = (manifest as any)[field];
-				if (!v) continue;
+		const categoryFilterList = ['appearance', 'developer tools', 'productivity', 'themes', 'integrations', 'viewer', 'search', 'tags', 'editor', 'files', 'personal knowledge management'];
```

Comment:
This list shouldn't be copied and pasted all over the codebase. The list of categories should be defined in only one place (and ideally as an enum, unless it's too complicated).